### PR TITLE
www: use tag manager and track events from data attributes

### PR DESCRIPTION
**Pull request by gillkyle**
## Description

After looking into cleaning up the event categories we're tracking, we settled on using Tag Manager to manage events when possible primarily, and then use custom events or outbound links if it can be done easier in code. 

_This is described a little more in #24203_

The analytics plugin is still included so that `trackCustomEvent` or `OutboundLink` can be used with our tracking code. There's also the ability to push variables into the tag manager `dataLayer` with both plugins included.

## Related Issues

#24203

### CODE REVIEW

File: www/src/components/navigation-mobile.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ const MobileNavItem = ({ linkTo, label, icon }) => (
     }}
     getProps={getProps}
     to={linkTo}
+    data-gtm-nav-category="Mobile Footer"
```
Comment (tesseralis) [inline www/src/components/navigation-mobile.js:39]: Are you sure that the tag gets propagated to the underlying `<a>` from the `<Link>`? Maybe do inspect element and check that the tag appears?

**Comment by muescha** [comment]
suggestion for a followup PR: update the docs with more details about the tag manager (for example docs/adding-analytics.md)

File: www/src/components/navigation-mobile.js (hunk 1 above)
Comment (gillkyle) [reply www/src/components/navigation-mobile.js:39]: image

Here's the successful build on Gatsby Cloud with the a tags getting the data attribute 🙂

**Comment by gillkyle** [comment]
@muescha we do have the guide https://www.gatsbyjs.org/docs/adding-analytics/ that covers some general concepts, mostly about Google Analytics, adding some context about Tag Manager could indeed be helpful there. However, the changes of this PR are mostly to install and set up the plugin.

**Comment by LekoArts** [comment]
@gillkyle What's the status on this?

**Comment by gillkyle** [comment]
This could have helped clean up some of the data we've been collecting the last couple months but isn't too drastic a change. This relies on Tag Manager to fire events in many cases which we'll likely try and rely less on in the future as our strategy around what we track also changes in the future so I'll just close this and can revisit it when we look at tracking in more depth.

**Pull request closed without merging**